src/Integration/LocalServices/SolutionBindingInformationProvider.cs

```diff
@@ -31,13 +31,46 @@ public SolutionBindingInformationProvider(IServiceProvider serviceProvider)
         }
 
         #region ISolutionBindingInformationProvider
+        public bool IsSolutionBound()
+        {
+            return this.GetSolutionBinding() != null;
+        }
+
         public IEnumerable<Project> GetBoundProjects()
         {
+            return this.GetBoundProjects(this.GetSolutionBinding());
+        }
+
+        public IEnumerable<Project> GetUnboundProjects()
+        {
+            return this.GetUnBoundProject(this.GetSolutionBinding());
```

Comment:
To avoid reading the binding information more than once i.e. GetUnbound will use GetBound.
This service is called on the UI thread so optimizing disk reads is something I think worth doing.